JamGame/Assets/Scripts/TileUnion/MeetingRoomLogics.cs

```diff
@@ -0,0 +1,163 @@
+﻿using Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using TileUnion.Tile;
+using UnityEngine;
+
+namespace TileUnion
+{
+    [AddComponentMenu("Scripts/TileUnion.MeetingRoomLogics")]
+    public class MeetingRoomLogics : MonoBehaviour
+    {
+        [SerializeField]
+        private TileUnionImpl tileUnion;
+
+        public TileUnionImpl TileUnion => tileUnion;
+
+        [SerializeField]
+        private List<TileImpl> tilesToAdd;
+
+        public List<TileImpl> TilesToAdd => tilesToAdd;
+
+        [SerializeField]
+        private Direction growDirection;
+        public Direction GrowDirection => growDirection;
+
+        [SerializeField]
+        private int maximumSize;
+
+        [SerializeField]
+        [InspectorReadOnly]
+        private int currentSize = 2;
+
+        public int CurrentSize
+        {
+            get => currentSize;
+            set => currentSize = value;
+        }
+
+        [SerializeField]
+        [InspectorReadOnly]
+        private int employeePerGrow = 2;
+
+        [SerializeField]
+        private List<string> incorrectMarks = new();
+        public IEnumerable<string> IncorrectMarks => incorrectMarks;
+
+        public bool IsEnoughPlace(int employeeCount)
+        {
+            return employeeCount <= (currentSize * employeePerGrow) - 1;
+        }
+
+        public bool IsCanFitEmployees(int employeeCount)
+        {
+            return employeeCount <= (maximumSize * employeePerGrow) - 1;
+        }
+
+        public int GetGrowCountForFitEmployees(int employeeCount)
+        {
+            if (IsEnoughPlace(employeeCount))
+            {
+                return 0;
+            }
+            else
+            {
+                int need = employeeCount - ((currentSize * employeePerGrow) - 1);
+                return (need / employeePerGrow) + ((need % employeePerGrow) > 0 ? 1 : 0);
+            }
+        }
+
+        public struct MeetingRoomGrowingInformation
+        {
+            public IEnumerable<Vector2Int> MovingTileUnionPositions;
+            public IEnumerable<Vector2Int> PositionsToTake;
+            public Vector2Int MovingDirection;
+            public int MovingSteps;
+            public TileUnionImpl TargetTileUnion;
```

Comment:
```suggestion
            IEnumerable<Vector2Int> MovingTileUnionPositions;
            public IEnumerable<Vector2Int> PositionsToTake;
            Vector2Int MovingDirection;
            int MovingSteps;
            TileUnionImpl TargetTileUnion;
```

Reply:
And review what you're actually using, for example I don't see where you use `TargetTileUnion`